### PR TITLE
Fix harverster error part 2

### PR DESCRIPTION
**Pull request by alexduf**
While fixing this issue I realised our definition of what is `platform` has reached a confusing state. In fact I believe that confusion is what caused this bug in the first place.

I've created a ticket to address that MSS-1539.

Regarding this PR, I introduce a new concept `WorkerSqs` which points to each of our sender workers.
Then I ensured all the routing logic was encapsulated in a single pattern match (see `routeToSqs`) such that we guarantee that a single message won't match two workers.

✅  Tested on CODE

### CODE REVIEW

File: notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala [hunk 1]
```diff
@@ -60,18 +68,28 @@ trait HarvesterRequestHandler extends Logging {
 
   }
 
+  def routeToSqs: PartialFunction[HarvestedToken, (WorkerSqs, HarvestedToken)] = {
```
Comment (jacobwinch) [inline notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala:71]: This is nice and clear 💯

File: notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala [hunk 2]
```diff
@@ -60,18 +68,28 @@ trait HarvesterRequestHandler extends Logging {
...
   def queueShardedNotification(shardedNotifications: Stream[IO, ShardedNotification]): Stream[IO, Unit] = {
     for {
       shardedNotification <- shardedNotifications
-      androidSink = platformSink(shardedNotification, Android, androidLiveDeliveryService)
-      iosSink = platformSink(shardedNotification, Ios, iosLiveDeliveryService)
-      newsstandSink = platformSink(shardedNotification, Newsstand, iosEditionDeliveryService)
-      androidEditionSink = platformSink(shardedNotification, AndroidEdition, androidEditionDeliveryService)
-      iosEditionSink = platformSink(shardedNotification, IosEdition, iosEditionDeliveryService)
+      androidSink = platformSink(shardedNotification, Android, WorkerSqs.AndroidWorkerSqs, androidLiveDeliveryService)
+      androidBetaSink = platformSink(shardedNotification, AndroidBeta, WorkerSqs.AndroidBetaWorkerSqs, androidLiveDeliveryService)
```
Comment (jacobwinch) [inline notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala:83]: Should this be `androidBetaDeliveryService`?

Comment (alexduf) [reply notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Harvester.scala:83]: I'm glad you spotted that one!

**Comment by alexduf** [comment]
Awaiting client side investigation before merging this change. cc @mohammad-haque

**Pull request merged**